pkg/grpc/api_test.go

```diff
@@ -1206,6 +1206,117 @@ func TestSaveState(t *testing.T) {
 	}
 }
 
+func TestSaveStateContentType(t *testing.T) {
+	fakeStore := &daprt.MockStateStore{}
+
+	contentTypeProperty := "contentType"
+
+	var expectedContentType *string
```

Comment:
Please don't use variables like these that are modified in the scope outside of the function checking them. It makes the code much harder to read.

Would recommend instead attaching a `fakeStore.On()` in each test, so it's clearer what's being tested, and it'll be easier to modify this in the future.

Reply:
Reworked the tests to avoid that.